### PR TITLE
move pytube to ytdlp

### PR DESCRIPTION
**Pull request by ksyeo1010**

### CODE REVIEW

File: demo/youtube/octotube.py [hunk 1]
```diff
@@ -70,7 +75,8 @@ def main():
     parser.add_argument('--work-folder', default=os.path.expanduser('~/'))
     args = parser.parse_args()
 
-    webm_path = download(url=args.url, folder=args.work_folder)
+    webm_path = download_ytdlp(url=args.url, output_dir=args.work_folder)[0]
+    print(webm_path)
```
Comment (mrrostam) [inline demo/youtube/octotube.py:79]: is this `print` intentional or left over from debugging?

File: demo/youtube/octotube.py [hunk 2]
```diff
@@ -14,9 +14,10 @@
 import time
 from argparse import ArgumentParser
 from threading import Thread
+from typing import *
 
 import pvoctopus
-from pytube import YouTube
+from yt_dlp import YoutubeDL
```
Comment (mrrostam) [inline demo/youtube/octotube.py:20]: Now that you added type hinting, can you do it was well for the constructor of `ProgressAnimation`?

File: demo/youtube/octotube.py [hunk 3]
```diff
@@ -48,17 +49,21 @@ def stop(self):
...
-
-    return webm_path
+def download_ytdlp(url: str, output_dir: str, options: Optional[Dict[str, Any]] = None) -> List[str]:
+    ydl_opts = {
+        'outtmpl': "%(id)s.%(ext)s",
+        'format': 'bestaudio',
+        'paths': {
+            'home': output_dir
+        },
+        'geo_bypass': True,
```
Comment (mrrostam) [inline demo/youtube/octotube.py:59]: doesn't it need `'extract_audio': True`?

Comment (ksyeo1010) [reply demo/youtube/octotube.py]: this downloads audio only

File: demo/youtube/octotube.py [hunk 4]
```diff
@@ -48,17 +49,21 @@ def stop(self):
...
-    if not os.path.exists(webm_path):
-        anime = ProgressAnimation('Downloading %s' % url)
-        anime.start()
-        youtube = YouTube(url)
-        audio_stream = youtube.streams.filter(only_audio=True, audio_codec='opus').order_by('bitrate').last()
-        audio_stream.download(output_path=folder, filename=os.path.basename(webm_path), skip_existing=True)
-        anime.stop()
-
-    return webm_path
+def download_ytdlp(url: str, output_dir: str, options: Optional[Dict[str, Any]] = None) -> List[str]:
```
Comment (mrrostam) [inline demo/youtube/octotube.py:52]: Can you keep the old interface? This function is only used here and doesn't need to be fancy

```
def download(url: str, output_dir: str) -> str:
```

**Review by mrrostam — APPROVED** [review]

**Pull request merged**